### PR TITLE
[backoff] cap initial backoff at max backoff

### PR DESCRIPTION
**Pull request by markdroth**

### CODE REVIEW

**Comment by ctiller** [comment]
Let's add an experiment here: this code is used fairly broadly through our codebase, and there are several places we take configuration from external sources -- meaning if anyone happens to rely on the current behavior they could see a spike in load soon after it changes.

**Comment by markdroth** [comment]
Added an experiment.

File: test/core/util/backoff_test.cc [hunk 1]
```diff
@@ -50,6 +50,24 @@ TEST(BackOffTest, ConstantBackOff) {
   EXPECT_EQ(backoff.NextAttemptDelay(), kInitialBackoff);
 }
 
+TEST(BackOffTest, InitialBackoffCappedByMaxBackoff) {
+  const auto kInitialBackoff = Duration::Seconds(2);
```
Comment (ctiller) [inline test/core/util/backoff_test.cc:54]: Expect this needs a `if (!IsBackoffCapInitialAtMaxEnabled()) GTEST_SKIP()`

**Review by ctiller — APPROVED** [review]

File: test/core/util/backoff_test.cc (hunk 1 above)
Comment (markdroth) [reply test/core/util/backoff_test.cc:54]: In experiments.yaml, I didn't list any tests to be executed without the experiment, so doesn't that mean this won't be run that way?

Comment (markdroth) [reply test/core/util/backoff_test.cc:58]: Looks like you're right, this is needed to import, since the experiment isn't yet enabled internally.

**Pull request closed without merging**